File: sdks/typescript/examples/quickstart.ts
/**
 * Quickstart: create a sandbox, list, destroy.
 *
 * Run with:  npx tsx quickstart.ts
 */
import createClient from 'openapi-fetch'
import type { paths } from '../src'

const client = createClient<paths>({
  baseUrl: 'http://localhost:8000',
  headers: { Authorization: 'Bearer hk_xxxxxxxxxxxx' },
})

const { data: sb, error } = await client.POST('/api/sandbox', {
  body: { snapshot_id: 'alpine:3.20', cpu: 1, memory_mb: 512 },
})
if (error) throw new Error(JSON.stringify(error))
console.log(`Created sandbox ${sb!.id} state=${sb!.state}`)

const { data: list } = await client.GET('/api/sandbox')
console.log(`Total sandboxes: ${list?.length}`)

await client.DELETE('/api/sandbox/{sandbox_id}', {
  params: { path: { sandbox_id: sb!.id } },
})
console.log(`Destroyed ${sb!.id}`)

File: sdks/python/examples/quickstart.py
"""Quickstart: spawn a sandbox via husk-client."""

from husk_client import AuthenticatedClient
from husk_client._generated.api.sandbox import (
    create_sandbox_api_sandbox_post,
    delete_sandbox_api_sandbox__sandbox_id__delete,
    list_sandboxes_api_sandbox_get,
)
from husk_client._generated.models import CreateRequest

client = AuthenticatedClient(
    base_url="http://localhost:8000",
    token="hk_xxxxxxxxxxxx",
)

sb = create_sandbox_api_sandbox_post.sync(
    client=client,
    body=CreateRequest(snapshot_id="alpine:3.20", cpu=1, memory_mb=512),
)
print(f"Created sandbox {sb.id} state={sb.state}")

all_sandboxes = list_sandboxes_api_sandbox_get.sync(client=client)
print(f"Total sandboxes: {len(all_sandboxes)}")

delete_sandbox_api_sandbox__sandbox_id__delete.sync_detailed(client=client, sandbox_id=sb.id)
print(f"Destroyed {sb.id}")
